sylt-compiler/src/typechecker.rs

```diff
@@ -777,10 +782,10 @@ impl TypeChecker {
                     None => {}
                 };
                 let ty = match self.globals.get(&(ctx.namespace, enum_name.name.clone())) {
-                    Some(Name::Enum(ty)) => {
-                        // NOTE(ed): This allows enumerations to be generic, which is always fun!
-                        let ty = ty.clone();
-                        let ty = self.push_type(ty);
+                    Some(Name::Type(ty)) => {
+                        // NOTE(ed): This allows generics, which is always fun!
+                        let ty = *ty;
+                        let ty = self.copy(ty);
```

Comment:
```suggestion
                        let ty = self.copy(*ty);
```
It is just a usize right?

Reply:
I tried this - it doesn't work since it's a borrow from self and Rust gets sad and sucks.